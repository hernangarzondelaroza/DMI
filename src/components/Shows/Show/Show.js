import React, { useEffect } from "react";

import { Card, Button, Row } from "react-bootstrap";
import { Link, Outlet } from "react-router-dom";

import { noImage } from "../../../utils/utils";

import PropTypes from "prop-types";

import "./Show.scss";

const Show = ({ data }) => {
  useEffect(() => {
    console.log(data)
  }, [data]);

  return (
    <>
      <Row as="section" className="mt-5">
        {data.map((element, key) => (
          <Card key={element?.show?.id} className="showCard">
            <Card.Img
              className="showImage"
              variant="top"
              src={element?.show?.image?.medium || { noImage }}
            />
            <Card.Body>
              <Card.Title>{element?.show?.name}</Card.Title>
              <div
                dangerouslySetInnerHTML={{
                  __html: element?.show?.summary || "No description provided"
                }}
                className="showSummary"
              />
              <Button variant="primary" className="cardButton">
                <Link
                  to={`/details/${element?.show?.id}`}
                  key={element?.show?.id}
                >
                  Show details
                </Link>
              </Button>
            </Card.Body>
          </Card>
        ))}
        <Outlet />
      </Row>
    </>
  );
};

Show.propTypes = {
  data: PropTypes.array
};

export default Show;
