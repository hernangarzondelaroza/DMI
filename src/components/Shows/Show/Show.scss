.showCard:first-of-type {
  margin-top: 0;
}

.showCard {
  margin: 1.25rem auto;
  padding: 1.25rem;
}

.showImage {
  margin: 0 auto;
  width: 9.375rem;
  border-radius: 50%;
  box-shadow: 0 0.188rem 0.625rem rgb(0 0 0 / 0.2);
}

.showSummary p,
.showSummary div {
  color: black;
}

.cardButton a {
  color: white;
  text-decoration: none;
}
