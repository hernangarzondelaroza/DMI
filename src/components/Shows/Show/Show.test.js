import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom'
import Show from './Show';

const mockedData = [
  {
    "show": {
      "image": {
        "medium": "https://static.tvmaze.com/uploads/images/medium_portrait/31/78286.jpg",
        "original": "https://static.tvmaze.com/uploads/images/original_untouched/31/78286.jpg"
      },
      "name": "Girls",
      "summary": "Lorem ipsum...",
      "id": 139,

    }
  }
]

describe('Show', () => {
  it('Should take a snapshot', () => {
    const component = renderer
      .create(<MemoryRouter><Show data={mockedData} /></MemoryRouter>);
    expect(component).toMatchSnapshot();
  })
})