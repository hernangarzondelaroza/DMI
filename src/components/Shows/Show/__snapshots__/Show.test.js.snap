// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Should take a snapshot 1`] = `
<section
  className="mt-5 row"
>
  <div
    className="showCard card"
  >
    <img
      className="card-img-top showImage"
      src="https://static.tvmaze.com/uploads/images/medium_portrait/31/78286.jpg"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Girls
      </div>
      <div
        className="showSummary"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Lorem ipsum...",
          }
        }
      />
      <button
        className="cardButton btn btn-primary"
        disabled={false}
        type="button"
      >
        <a
          href="/details/139"
          onClick={[Function]}
        >
          Show details
        </a>
      </button>
    </div>
  </div>
</section>
`;
